static/extensions/Ikelene/email2FA.js

```diff
@@ -0,0 +1,317 @@
+                            },
+                            PROJECT_ID: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: 'abc123'
+                            }
+                        }
+                    },
+                    {
+                        opcode: 'isServerOnline',
+                        blockType: Scratch.BlockType.BOOLEAN,
+                        text: 'Is 2FA server online?'
+                    },
+                    {
+                        blockType: Scratch.BlockType.LABEL,
+                        text: 'Customization'
+                    },
+                    {
+                        opcode: 'setEmailPreset',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Set email style preset to [PRESET]',
+                        arguments: {
+                            PRESET: {
+                                type: Scratch.ArgumentType.STRING,
+                                menu: 'presetMenu'
+                            }
+                        }
+                    },
+                    {
+                        opcode: 'setCustomHTML',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'Set custom email HTML to [HTML]',
+                        arguments: {
+                            HTML: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: '<h1>Your Code</h1>'
+                            }
+                        }
+                    }
+                    //{
+                    //    blockType: Scratch.BlockType.LABEL,
+                    //    text: 'Use your own email'
+                    //},
+                    //{
+                    //    opcode: 'setCustomEmailLogin',
+                    //    blockType: Scratch.BlockType.COMMAND,
+                    //    text: 'Use custom email login [EMAIL] pass [PASS] smtp [SMTP] port [PORT]',
+                    //    arguments: {
+                    //        EMAIL: { type: Scratch.ArgumentType.STRING, defaultValue: 'your@email.com' },
+                    //        PASS: { type: Scratch.ArgumentType.STRING, defaultValue: 'password' },
+                    //        SMTP: { type: Scratch.ArgumentType.STRING, defaultValue: 'smtp.gmail.com' },
+                    //        PORT: { type: Scratch.ArgumentType.STRING, defaultValue: '465' }
+                    //    }
+                    //},
+                    //{
+                    //    opcode: 'clearCustomEmailLogin',
+                    //    blockType: Scratch.BlockType.COMMAND,
+                    //    text: 'Clear custom email login'
+                    //}
+                ],
+                menus: {
+                    presetMenu: {
+                        acceptReporters: true,
+                        items: [
```

Comment:
Could this be generated by the server. Like having the contents of this menu dynamically gotten from the server. That could be done by setting this (`items`) to a string with the same name as the function that returns an array for this list. This could allow for the dynamic adding of templates to your server without needing to update your extension.

Reply:
This would be especially nice if you ever decide to open source the backend, since it would allow for people to fork your backend to add more email templates without needing to create a fork of the extension (which is essentially the whole goal of allowing for the user to change the server).